cmd/utils/flags.go

```diff
@@ -1845,43 +1844,26 @@ func RegisterRaftService(stack *node.Node, ctx *cli.Context, nodeCfg *node.Confi
 	useDns := ctx.GlobalBool(RaftDNSEnabledFlag.Name)
 	raftPort := uint16(ctx.GlobalInt(RaftPortFlag.Name))
 
+	var myId uint64
+	var joinExisting bool
+
 	if err := stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
 		privkey := nodeCfg.NodeKey()
 		strId := enode.PubkeyToIDV4(&privkey.PublicKey).String()
 		blockTimeNanos := time.Duration(blockTimeMillis) * time.Millisecond
 		peers := nodeCfg.StaticNodes()
 
-		var myId uint16
-		var joinExisting bool
-
 		if joinExistingId > 0 {
-			myId = uint16(joinExistingId)
+			myId = uint64(joinExistingId)
 			joinExisting = true
 		} else if len(peers) == 0 {
-			Fatalf("Raft-based consensus requires either (1) an initial peers list (in static-nodes.json) including this enode hash (%v), or (2) the flag --raftjoinexisting RAFT_ID, where RAFT_ID has been issued by an existing cluster member calling `raft.addPeer(ENODE_ID)` with an enode ID containing this node's enode hash.", strId)
-		} else {
-			peerIds := make([]string, len(peers))
-
-			for peerIdx, peer := range peers {
-				if !peer.HasRaftPort() {
-					Fatalf("raftport querystring parameter not specified in static-node enode ID: %v. please check your static-nodes.json file.", peer.String())
-				}
-
-				peerId := peer.ID().String()
-				peerIds[peerIdx] = peerId
-				if peerId == strId {
-					myId = uint16(peerIdx) + 1
-				}
-			}
-
-			if myId == 0 {
-				Fatalf("failed to find local enode ID (%v) amongst peer IDs: %v", strId, peerIds)
-			}
+			Fatalf("Raft-based consensus requires either an initial peers list including all the nodes in the network (in static-nodes.json) including this enode hash (%v)", strId)
```

Comment:
this is no longer an either-or statement. Suggest "requires an initial peers list"